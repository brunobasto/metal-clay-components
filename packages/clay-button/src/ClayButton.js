'use strict';

import ClayButtonBase from './ClayButtonBase';
import Soy from 'metal-soy';
import { Config } from 'metal-state';

import 'clay-icon';

import templates from './ClayButton.soy.js';

/**
 * Metal Clay Button component.
 */
class ClayButton extends ClayButtonBase {}

/**
 * State definition.
 * @static
 * @type {!Object}
 */
ClayButton.STATE = {
	/**
	 * Aria label attribute for the link element.
	 * @instance
	 * @memberof ClayButton
	 * @type {string}
	 * @default ''
	 */
	ariaLabel: Config.string(),

	/**
	 * Renders the button as a block element.
	 * @instance
	 * @memberof ClayButton
	 * @type {boolean}
	 * @default false
	 */
	block: Config.bool().value(false),

	/**
	 * Render ClayIcon in the ClayButton element, available options are
	 * `elementClasses`, `spritemap`, `symbol`.
	 * @instance
	 * @memberof ClayButton
	 * @type {?Object|undefined}
	 * @default undefined
	 */
	icon: Config.object(),

	/**
	 * The name that will make the CSS class name that will represent the
	 * button size.
	 * @instance
	 * @memberof ClayButton
	 * @type {?string|undefined}
	 * @default undefined
	 */
	size: Config.string(),

	/**
	 * The css class that colors the button.
	 * @instance
	 * @memberof ClayButton
	 * @type {?string|undefined}
	 * @default undefined
	 */
	style: Config.string(),
};

Soy.register(ClayButton, templates);

export { ClayButton };
export default ClayButton;
