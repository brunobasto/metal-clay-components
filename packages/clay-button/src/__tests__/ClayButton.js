'use strict';

import ClayButton from '../ClayButton';

let button;

describe('ClayButton', function() {
	afterEach(() => {
		if (button) {
			button.dispose();
		}
	});

	it('should generate the default markup', function() {
		button = new ClayButton();

		expect(button).toMatchSnapshot();
	});

	it('should render a submit button', function() {
		button = new ClayButton({
			type: 'submit',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a disabled button', function() {
		button = new ClayButton({
			disabled: true,
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a success button', function() {
		button = new ClayButton({
			style: 'success',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a button with label', function() {
		button = new ClayButton({
			label: 'Label',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a button with label', function() {
		button = new ClayButton({
			name: 'myButton',
		});

		expect(button).toMatchSnapshot();
	});

	it('should render a link with ariaLabel attribute', function() {
		button = new ClayButton({
			ariaLabel: 'My Description',
			href: 'http://liferay.com',
			label: 'Visit Liferay.com',
		});

		expect(button).toMatchSnapshot();
	});
});
