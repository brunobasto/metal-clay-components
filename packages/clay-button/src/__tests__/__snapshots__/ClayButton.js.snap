// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayButton should generate the default markup 1`] = `<button class="btn btn-primary"></button>`;

exports[`ClayButton should render a button with label 1`] = `<button class="btn btn-primary">Label</button>`;

exports[`ClayButton should render a button with label 2`] = `<button class="btn btn-primary" name="myButton"></button>`;

exports[`ClayButton should render a disabled button 1`] = `<button class="btn btn-primary" disabled=""></button>`;

exports[`ClayButton should render a link with ariaLabel attribute 1`] = `<a href="http://liferay.com" class="btn btn-primary" aria-label="My Description">Visit Liferay.com</a>`;

exports[`ClayButton should render a submit button 1`] = `<button class="btn btn-primary" type="submit"></button>`;

exports[`ClayButton should render a success button 1`] = `<button class="btn btn-success"></button>`;
